infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -132,7 +138,7 @@ synchronized void dispatch(@Nonnull IWorkflow workflow,
             activeWorkflows.put(workflow.getId(), req.getEndpoint());
 
             CompletableFuture.runAsync(() -> {
-                run(workflow);
+                run(workflow, req.getActionRetry());
```

Comment:
Why is this being put in a completableFuture if the result is never checked?

Reply:
I've change it to ForkJoinPool.commonPool().execute

Reply:
You shouldn't use the common pool. Create a new thread and set the name to base64 of the workflow id